.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,151 @@
+# This workflow builds 
+# - source distribution (SDist)
+# - wheels using cibuildwheels (https://iscinumpy.dev/tags/cibuildwheel/)
+# and upload to PyPI (https://github.com/marketplace/actions/pypi-publish)
+
+name: Build_with_cibuildwheel_and_upload_to_PyPI
+
+
+# The workflow triggers on:
+
+#  - workflow_dispatch - manually trigger the workflow run using the GitHub browser interface 
+# see https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/
+# https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
+
+# - push to 'master', only if changes are made to 'MCEq' directory (excluding MCEq/data/README.md)
+# see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches-and-tags
+
+# - on pull request opened on a 'master' branch when a pull request that includes 
+# a change to 'MCEq' directory
+# see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+
+on:
+  workflow_dispatch:
+    # inputs:
+    #   artifactNameSDist:
+    #     description: 'Name of artifact for SDist'
+    #     required: true
+    #     default: 'SDist'
+    #   artifactNameWheels:
+    #     description: 'Prefix for artifacts for wheels'
+    #     required: true
+    #     default: 'wheels'
+  push:
+    branches:
+      - master
+    paths:
+      - 'MCEq/**'
+      - '!MCEq/data/README.md'
+    tags:        
+      - 'release**' 
+      # push tag release** (release_..., release/...) initiates upload to PyPI
+      # see also https://stackoverflow.com/a/71076957
+  pull_request:
+    types:
+      - opened
+    branches:    
+      - master
+    paths:
+      - 'MCEq/**'
+      - '!MCEq/data/README.md'
+
+jobs:
+    make_sdist:
+      name: Make SDist
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v3
+          # with:
+          # # number of commits to fetch, default: "1". "0" means all history
+          #   fetch-depth: 0
+
+        - name: Build SDist
+          run: pipx run build --sdist
+
+        - uses: actions/upload-artifact@v3
+          with:
+          # name: ${{ github.event.inputs.artifactNameSDist }}
+          # If "name" is not provided, "artifact" will be used as the default name
+            path: dist/*.tar.gz
+            
+    build_wheels:
+      name: Wheel on ${{ matrix.os }}
+      runs-on: ${{ matrix.os }}
+      strategy:
+        fail-fast: false
+        matrix:
+          os: [ubuntu-20.04, windows-2019, macos-10.15]
+
+      steps:
+      - uses: actions/checkout@v3
+        
+        # Building non-native architectures using emulation
+        # https://cibuildwheel.readthedocs.io/en/stable/faq/
+      - name: Set up QEMU
+        if: runner.os == 'Linux'
```

Comment:
Doesn’t it mean that all linux runners will go through QEMU?

Reply:
We install QEMU static binaries for the runner if it is Linux. In our case it will be once for ubuntu-20.04. After setting up QEMU and setting CIBW_ARCHS_LINUX environment variable "cibuildwheel" will automatically use emulatation for the required architectures. In configuration file "CIBW_ARCHS_LINUX: auto aarch64" is set. "auto" is for native architecture.